astromodels/functions/template_model.py

```diff
@@ -428,68 +518,64 @@ def _custom_init_(self, model_name, other_name=None, log_interp=True):
 
         # Get the data directory
 
-        data_dir_path = get_user_data_path()
+        data_dir_path: Path = get_user_data_path()
 
         # Sanitize the data file
 
-        filename_sanitized = os.path.abspath(
-            os.path.join(data_dir_path, "%s.h5" % model_name)
-        )
+        filename_sanitized: Path = data_dir_path.absolute() /  f"{model_name}.h5"
+        
+        if not filename_sanitized.exists():
 
-        if not os.path.exists(filename_sanitized):
-
-            raise MissingDataFile(
-                "The data file %s does not exists. Did you use the "
-                "TemplateFactory?" % (filename_sanitized)
-            )
+            log.error("The data file %s does not exists. Did you use the "
+                "TemplateFactory?" % (filename_sanitized))
+            
+            raise MissingDataFile()
 
         # Open the template definition and read from it
 
         self._data_file = filename_sanitized
 
-        with HDFStore(filename_sanitized) as store:
-
-            self._data_frame = store["data_frame"]
-
-            self._parameters_grids = collections.OrderedDict()
+        # use the file shadow to read
+        
+        template_file: TemplateFile = TemplateFile.from_file(filename_sanitized)
 
-            processed_parameters = 0
-
-            for key in list(store.keys()):
-
-                match = re.search("p_([0-9]+)_(.+)", key)
-
-                if match is None:
+        self._parameters_grids = collections.OrderedDict()
 
-                    continue
+        processed_parameters = 0
 
-                else:
+        for key in template_file.parameter_order:
 
-                    tokens = match.groups()
+            try:
 
-                    this_parameter_number = int(tokens[0])
-                    this_parameter_name = str(tokens[1])
+                # sometimes this is
+                # stored binary
+                
+                k = key.decode()
 
-                    assert (
-                        this_parameter_number == processed_parameters
-                    ), "Parameters out of order!"
+            except(AttributeError):
 
-                    self._parameters_grids[this_parameter_name] = store[key]
+                # if not, then we
+                # load as a normal str
+                
+                k = key
+                                
+            log.debug(f"reading parameter {str(k)}")
 
-                    processed_parameters += 1
+  
+            self._parameters_grids[str(k)] = template_file.parameters[key]
 
-            self._energies = np.array(store["energies"])
+            processed_parameters += 1
```

Comment:
I think you can delete this processed_parameters variable? I don't see where it is used.

Reply:
done!